src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPath.cs

```diff
@@ -117,7 +118,7 @@ IEnumerator IEnumerable.GetEnumerator()
         [SuppressMessage("Microsoft.Design", "CA1011:ConsiderPassingBaseTypesAsParameters", Justification = "We would rather ship the PathSegmentTranslator so that its more extensible later")]
         public IEnumerable<T> WalkWith<T>(PathSegmentTranslator<T> translator)
         {
-            return this.segments.Select(segment => segment.TranslateWith(translator));
+            return this.Select(segment => segment.TranslateWith(translator));

```

Comment:
Don't we need `return this.segments.Select(segment => segment.TranslateWith(translator));` here?

Reply:
Nice catch John, I had missed this when i was reverting from the original change with List